compiler-rt/test/ctx_profile/Unit/lit.site.cfg.py.in

```diff
@@ -21,7 +21,11 @@ config.test_source_root = config.test_exec_root
 # When LLVM_ENABLE_PER_TARGET_RUNTIME_DIR=on, the initial value of
 # config.compiler_rt_libdir (COMPILER_RT_RESOLVED_LIBRARY_OUTPUT_DIR) has the
 # host triple as the trailing path component. The value is incorrect for i386
-# tests on x86_64 hosts and vice versa. But, since only x86_64 is enabled as
-# target, and we don't support different environments for building and,
-# respectivelly, running tests, we shouldn't see this case.
-assert not config.enable_per_target_runtime_dir or config.target_arch == config.host_arch
+# tests on x86_64 hosts and vice versa. Adjust config.compiler_rt_libdir
+# accordingly.
+if config.enable_per_target_runtime_dir and config.target_arch != config.host_arch:
```

Comment:
I would prefer not having the general handling of asan, because we don't mean to be here for 32 bit anyway. so if `config.host_arch` is not set in your case, what's the rest - I assume config.target_arch is exactly x86_64 and then we could assert that and do the behavior on line 30?

Reply:
> I would prefer not having the general handling of asan, because we don't mean to be here for 32 bit anyway.
Fair.

> so if `config.host_arch` is not set in your case, what's the rest - I assume config.target_arch is exactly x86_64 and then we could assert that and do the behavior on line 30?

The config.target_arch is set as expected.  There is already an assert that its X86_64 on line 14. So maybe just

```
if config.enable_per_target_runtime_dir:
  ...
```

Reply:
ah, but that's only for the CTX_PROFILE.

Maybe a better approach is to assert it isn't i386? Then you have the ASAN like fixup on the libdir and 32-bit is still unsupported

Reply:
sgtm. we have an assert earlier. OK.

Reply:
Since we have the assert in both ctx_profile and memprof, I think this simplified check is appropriate.